server/src/main/java/org/elasticsearch/cluster/metadata/MetadataDataStreamsService.java

```diff
@@ -91,6 +142,38 @@ static ClusterState modifyDataStream(
         return ClusterState.builder(currentState).metadata(updatedMetadata).build();
     }
 
+    /**
+     * Creates an updated cluster state in which the requested data streams have the data lifecycle provided.
+     * Visible for testing.
+     */
+    static ClusterState modifyDataLifecycle(
+        ClusterState currentState,
+        List<String> dataStreamNames,
+        @Nullable DataLifecycle dataLifecycle
+    ) {
+        Metadata updatedMetadata = currentState.metadata();
```

Comment:
This seems to be the current metadata right? 

The updated metadata is something we build at line 173 where we call ```
builder.build();
```
Shall we avoid the `updatedMetadata` reassignment?

Reply:
Good catch, fixed.